src/okareo/crewai_logger.py

```diff
@@ -0,0 +1,96 @@
+import datetime
+import json
+import random
+import string
+import traceback
+import uuid
+from typing import Any
+
+from opentelemetry import trace
+from opentelemetry.sdk.trace import ReadableSpan, SpanProcessor, TracerProvider
+
+from okareo import Okareo
+
+
+class CrewAISpanProcessor(SpanProcessor):
+    def __init__(self, config: dict[str, Any]) -> None:
+        if "api_key" not in config:
+            raise ValueError("api_key is required in the config")
+
+        self.okareo = Okareo(
+            config["api_key"],
+        )
+        self.session_id = str(config.get("context_token", uuid.uuid4()))
+        self.tags = config.get("tags", [])
+
+        random_suffix = "".join(
+            random.choices(string.ascii_lowercase + string.digits, k=5)
+        )
+        self.mut_name = config.get("mut_name", f"crewai-chat-{random_suffix}")
+
+        self.registered_model = self.okareo.register_model(
+            name=self.mut_name, tags=self.tags
+        )
+
+    def _format_span_data(self, span: ReadableSpan) -> dict:
+        if span.attributes is None:
+            return {}
+        return {
+            k: self.safe_loads(v)
+            for k, v in span.attributes.items()
+            if self.is_serializable(self.safe_loads(v))
+        }
+
+    def _get_timestamp_iso(self, epoch_ns: Any) -> str:
+        return datetime.datetime.fromtimestamp(
+            epoch_ns / 1e9, tz=datetime.timezone.utc
+        ).isoformat()
+
+    def on_end(self, span: ReadableSpan) -> None:
+        if span.instrumentation_info.name != "crewai.telemetry":
+            return
+
+        try:
+            span_data = self._format_span_data(span)
+            result_obj = {
+                "log_type": span.name,
+                "trace_id": format(span.context.trace_id, "032x"),
+                "span_id": format(span.context.span_id, "016x"),
+            }
+
+            self.registered_model.add_data_point(
+                input_obj=span_data,
+                input_datetime=self._get_timestamp_iso(span.start_time),
+                result_obj=json.dumps(result_obj),
+                result_datetime=self._get_timestamp_iso(span.end_time),
+                context_token=self.session_id,
+                tags=self.tags,
+            )
```

Comment:
I realize the impl differs a bit from the autogen impl, but I think it would be nice to print out some '[Okareo]' statements to let the user know that the logger is working. For example, the autogen logger prints the following when the logger is done:

> Next speaker: customer
>
> [Okareo] Finished logging Session ID: 5a9643b1-aefb-4b80-9e7c-2d0ac61fe599
> [Okareo] Logged data points for autogen chat under mut_name 'autogen-stock-chat-example-VSdul'.
> [Okareo] View data points at http://localhost:3000/project/cb832f0d-0e3d-4bed-9d9c-023441ec9dd5/model/cbd25609-30a1-420b-8caf-f1ecedafb32d.

Once we have the Group ID backend implementation, we can switch our app_link over to use the group ID